### ISSUE TITLE
Use zipfile.ZipFile.open instead of twisted.python.zipstream in twisted.python.zippath

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4498]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4498|
|Type|enhancement|
|Created|2010-06-08 17:58:29Z|

Python 2.6 introduces an `open` method on which provides file-like access to a file inside a zipfile.  This provides the streaming access necessary to implement `twisted.python.zippath.ZipPath.open`.

On older versions of Python, we'll still have to use `twisted.python.zipstream`, of course.

<details><summary>Searchable metadata</summary>

```
trac-id__4498 4498
type__enhancement enhancement
reporter__exarkun exarkun
priority__lowest lowest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__core core
keywords__easy easy
time__1276019909000000 1276019909000000
changetime__1317835971000000 1317835971000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph removed owner|
|-|-|

I don't think there's really much value in doing this; as far as I know, our implementations of this functionality are largely equivalent.  It would be nice to get rid of `zipstream` eventually, of course, but it's going to be years before we can drop 2.5.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

The main value in doing this now is probably to determine if the thing in the standard library actually works.  If we report bugs against it now, they might actually be fixed in a released version of Python by the time we _could_ drop our own implementation.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

I've done this in lp:filepath now.  It was actually a very simple change, just a couple lines, and the unit tests all still pass.

See http://bazaar.launchpad.net/~exarkun/filepath/trunk/revision/11#filepath/_zippath.py

(Note that this is without the fallback for older Pythons)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="aruse's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|aruse commented|
|-|-|

This appears to be a duplicate of [#4035](https://github.com/twisted/twisted/issues/4035), where I just submitted a patch.

**Comment by twisted-trac** [comment]
|[<img alt="bitglue's avatar" src="https://avatars.githubusercontent.com/u/1232472?s=50" width="50" height="50">](https://github.com/bitglue)|@bitglue set status to `closed`|
|-|-|
